qa/tasks/cephfs/test_misc.py

```diff
@@ -524,6 +524,30 @@ def test_session_ls(self):
     def test_client_ls(self):
         self._session_client_ls(['client', 'ls'])
         
+    
+class TestSessionClientEvict(CephFSTestCase):
+    CLIENTS_REQUIRED = 3
+
+    def _session_client_evict(self, cmd):
+        mount_a_client_id = self.mount_a.get_global_id()
+        info = self.fs.rank_asok(cmd + ['ls'])
+        self.assertIsNotNone(len(info), 3)
+        with self.assertRaises(CommandFailedError) as ce:
+            self.fs.rank_asok(cmd + ['evict'])
+        self.assertEqual(ce.exception.exitstatus, -errno.EPERM)
+        info = self.fs.rank_asok(cmd + ['evict', f'id={mount_a_client_id}'])
+        self.assertEqual(len(info), 2)
```

Comment:
Do you have some sort of output showing successful execution of this test?  From a quick look at the code, `evict` command shouldn't be producing any output -- `Formatter` isn't even passed to `evict_clients()`.  Combined with the bogus `assertIsNotNone` above, I'm having doubts that this is working at all.

Reply:
I am using `self.run_ceph_cmd()` here:https://github.com/ceph/ceph/pull/49974/files#diff-b26969958eeaeb10cb96e75c7b86b60ffbca19b73e0593eb5baa71a1045ab447R536 as `self.fs.rank_asok()` output demands json output which is not the case when client evict command fails.

Reply:
But the command itself is supposed to fail and `CommandFailedError` is supposed to be raised in that case, so it shouldn't matter?

Regardless, using different ways to run the same command for success tests and for failure tests is frowned upon because then it's not obvious whether failure tests are failing legitimately or just because of the different command wrapper.

Reply:
> > as `self.fs.rank_asok()` output demands json output which is not the case when client evict command fails.
> 
> But the command itself is supposed to fail and `CommandFailedError` is supposed to be raised in that case, so it shouldn't matter?
> 
> Regardless, using different ways to run the same command for success tests and for failure tests is frowned upon because then it's not obvious whether failure tests are failing legitimately or just because of the different command wrapper.

Let me be bit more explanative:
`self.fs.rank_asok()` uses `json_asok()` for execution of commands which runs successfully only when the output is in the form of JSON, which is not the case in this change.
This line https://github.com/ceph/ceph/blob/main/qa/tasks/cephfs/filesystem.py#L275 is responsible for that, which actually wants output in JSON format. So this is the reason why I switched from `self.fs.rank_asok()`. Currently the test uses `self.fs.rank_tell()`

Reply:
> > > as `self.fs.rank_asok()` output demands json output which is not the case when client evict command fails.
> > 
> > 
> > But the command itself is supposed to fail and `CommandFailedError` is supposed to be raised in that case, so it shouldn't matter?
> > Regardless, using different ways to run the same command for success tests and for failure tests is frowned upon because then it's not obvious whether failure tests are failing legitimately or just because of the different command wrapper.
> 
> Let me be bit more explanative: `self.fs.rank_asok()` uses `json_asok()` for execution of commands which runs successfully only when the output is in the form of JSON, which is not the case in this change. This line https://github.com/ceph/ceph/blob/main/qa/tasks/cephfs/filesystem.py#L275 is responsible for that, which actually wants output in JSON format. So this is the reason why I switched from `self.fs.rank_asok()`. Currently the test uses `self.fs.rank_tell()`

When `ceph daemon mds.b --format=json session evict` is executed; it returns:
```
ERROR: (1) Operation not permitted
pass --yes-i-really-mean-it to evict all client sessions
```
which is picked up by `response_data` in `json_asok()` and it needs to have a valid JSON schema else while decoding it would fail(the line @neesingh-rh mentioned above):
```
Traceback (most recent call last):
  File "<string>", line 12, in <module>
  File "/usr/local/lib/python3.11/json/__init__.py", line 359, in loads
return cls(**kw).decode(s)
^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/json/decoder.py", line 337, in decode
obj, end = self.raw_decode(s, idx=_w(s, 0).end())
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/json/decoder.py", line 355, in raw_decode
ERROR!
raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

`self.fs.rank_asok()` needs to be fixed to handle such cases

Reply:
FYI i just raised a tracker `rank_asok()` ->  https://tracker.ceph.com/issues/63176